lib/Word_Lib/Signed_Words.thy

```diff
@@ -93,6 +93,21 @@ lemma zero_sle_ucast:
   apply (simp_all add: bit_simps disjunctive_add)
   done
 
+lemma nth_w2p_scast:
+  "(bit (scast ((2::'a::len signed word) ^ n) :: 'a word) m)
+         \<longleftrightarrow> (bit (((2::'a::len  word) ^ n) :: 'a word) m)"
+  by (simp add: bit_simps)
+
+lemma scast_nop1 [simp]:
+  "((scast ((of_int x)::('a::len) word))::'a signed word) = of_int x"
+  by transfer (simp add: take_bit_signed_take_bit)
+
+lemma scast_nop2 [simp]:
+  "((scast ((of_int x)::('a::len) signed word))::'a word) = of_int x"
+  by transfer (simp add: take_bit_signed_take_bit)
```

Comment:
when these are proved by `transfer`, what is is transferring to here?

Reply:
I have no idea, these are all Florian's doing and just re-imported. I assume that the base type for this is `int` (because of the `of_int`, so `transfer` would use the corresponding theorems for `of_int` to produce an `int` goal from this.